.github/workflows/build.yaml

```diff
@@ -47,3 +61,143 @@ jobs:
+          sudo apt-get install --assume-yes --no-install-recommends \
+          libgstrtspserver-1.0-dev:arm64 \
+          libgstreamer1.0-dev:arm64 \
+          libgtk2.0-dev:arm64
+      - name: Install rustup arm64
+        uses: actions-rs/toolchain@v1
+        with:
+            toolchain: stable
+            override: true
+            target: aarch64-unknown-linux-gnu
+      - name: Build arm64
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --release --all-features --target=aarch64-unknown-linux-gnu
+        env:
+          CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER: aarch64-linux-gnu-gcc
+          CC_aarch64_unknown_linux_gnu: aarch64-linux-gnu-gcc
+          CXX_aarch64_unknown_linux_gnu: aarch64-linux-gnu-g++
+          PKG_CONFIG_ALLOW_CROSS: 1
+          PKG_CONFIG_PATH: /usr/lib/aarch64-linux-gnu/pkgconfig:/usr/share/pkgconfig/
+          PKG_CONFIG_LIBDIR: /usr/lib/aarch64-linux-gnu/pkgconfig:/usr/share/pkgconfig/
+      - uses: actions/upload-artifact@v2
+        with:
+          name: release-arm64-buster
+          path: "target/aarch64-unknown-linux-gnu/release/neolink*"
+
+  i386build:
+    name: build-i386
+    runs-on: ubuntu-latest
+    container: "node:current-buster-slim"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install basic tools
+        run: |
+          apt-get update
+          apt-get install --assume-yes --no-install-recommends sudo curl ca-certificates build-essential
+      - name: Add arm64 as a foreign architecture
+        run: |
+          sudo dpkg --add-architecture i386
+      - name: Install cross compile tools
+        run: |
+          sudo apt-get update
+          sudo apt-get install --assume-yes --no-install-recommends gcc-i686-linux-gnu
+      - name: Install i386 gstreamer
+        run: |
+          sudo apt-get update
+          sudo apt-get install --assume-yes --no-install-recommends \
+          libgstrtspserver-1.0-dev:i386 \
+          libgstreamer1.0-dev:i386 \
+          libgtk2.0-dev:i386
+      - name: Install rustup i386
+        uses: actions-rs/toolchain@v1
+        with:
+            toolchain: stable
+            override: true
+            target: i686-unknown-linux-gnu
+      - name: Build i386
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --release --all-features --target=i686-unknown-linux-gnu
+        env:
+          CARGO_TARGET_I686_UNKNOWN_LINUX_GNU_LINKER: i686-linux-gnu-gcc
+          CC_i686_unknown_linux_gnu: i686-linux-gnu-gcc
+          CXX_i686_unknown_linux_gnu: i686-linux-gnu-g++
+          PKG_CONFIG_ALLOW_CROSS: 1
+          PKG_CONFIG_PATH: /usr/lib/i386-linux-gnu/pkgconfig:/usr/share/pkgconfig/
+          PKG_CONFIG_LIBDIR: /usr/lib/i386-linux-gnu/pkgconfig:/usr/share/pkgconfig/
+      - uses: actions/upload-artifact@v2
+        with:
+          name: release-i386-buster
+          path: "target/i686-unknown-linux-gnu/release/neolink*"
```

Comment:
Is it possible to parameterize these into a single template (maybe "`ubuntu-cross-build`") that gets instantiated multiple times?

Reply:
I'm not sure about a template maybe a custom action? Not sure. I could do it using shell scripting instead. Maybe have a script that would take as input the arch and which then calls `cargo`. I think there are a few places we could reuse it in the packaging PR too.

I could also doit using some output variables generated in a prior action. But we are just moving the definitions elsewhere and I'm not sure it will help readability.

If I get the time I'll have a look into a generic build script that can be used with this and my packaging PR

Reply:
I'm talking about using [contexts and a matrix](https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions). If you don't mind, I'd like to take a shot at refactoring this. I would like to get more proficient at GH Actions.

Reply:
And, would you mind if I squash and rebase some of these commits?

Reply:
Squash and refractor away :).